Kernel/VirtIO/VirtIOConsole.h

```diff
@@ -1,52 +1,67 @@
 /*
- * Copyright (c) 2021, the SerenityOS developers.
+ * Copyright (c) 2021, Kyle Pereira <hey@xylepereira.me>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #pragma once
 
-#include <Kernel/Devices/CharacterDevice.h>
 #include <Kernel/VM/RingBuffer.h>
 #include <Kernel/VirtIO/VirtIO.h>
+#include <Kernel/VirtIO/VirtIOConsolePort.h>
 
 namespace Kernel {
+class VirtIOConsole
+    : public VirtIODevice
+    , public RefCounted<VirtIOConsole> {
```

Comment:
why is this refcounted?

Reply:
Ah wait this actually does have to be RefCounted, since the VirtIO class stores a RefPtr to it.

Reply:
Yes, but it inherits from a file, so i dont think it has to be explicitly refcounted as well.

Reply:
oh nevermind, you removed the inheritence from CharacterDevice, so it is required :)